lib/ruby-mpd/song.rb

```diff
@@ -24,6 +24,18 @@ def ==(another)
     self.class == another.class && self.file == another.file
   end
 
+  def to_json(*a)
+    {
+      "time" => @time,
+      "file" => @file,
+      "title" => @title,
+      "artist" => @artist,
+      "album" => @album,
+      "albumartist" => @albumartist,
+      "data" => @data
+    }.to_json(*a)
```

Comment:
Ah, I stand corrected, `to_json` is indeed valid then.

I would still refactor it by creating a `to_h` implementation and then simply derive the `to_json` implementation from that:

```ruby
def to_h
  {
    time: @time,
    file: @file,
    title: @title,
    artist: @artist,
    album: @album,
    albumartist: @albumartist,
  }.merge(data)
end

def to_json
  to_h.to_json
end
```